webapp/channels/src/components/root/root.tsx

```diff
@@ -273,39 +208,31 @@ export default class Root extends React.PureComponent<Props, State> {
             this.props.history.push('/signup_user_complete');
         }
 
-        Promise.all([
-            this.props.actions.initializeProducts(),
-            initializePlugins(),
-        ]).then(() => {
-            if (this.mounted) {
-                // supports enzyme tests, set state if and only if
-                // the component is still mounted on screen
-                this.setState({configLoaded: true});
-            }
-        });
+        await Promise.all([this.props.actions.initializeProducts(), initializePlugins()]);
+        this.setState({configLoaded: true});
```

Comment:
This is a slight functional change from before because we now wait for plugins/products to load and set `this.state.configLoaded` before applying the theme and such. Is that intended?

Also, this might potentially reintroduce the warnings about calling `setState` after the component is unmounted, but I don't think that's a practical concern at all. We know we're not leaking it, and I'm pretty sure I saw something about the React devs considering removing that warning

Reply:
You are correct this was brought in from another PR even there i have reverted this change